services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentServiceITest.kt

```diff
@@ -72,47 +76,68 @@ private const val ALLU_ID = 42
+    @Autowired private val applicationAttachmentService: ApplicationAttachmentService,
+    @Autowired private val applicationAttachmentRepository: ApplicationAttachmentRepository,
+    @Autowired private val contentRepository: ApplicationAttachmentContentRepository,
+    @Autowired private val alluDataFactory: AlluDataFactory,
+    @Autowired private val attachmentFactory: ApplicationAttachmentFactory,
+) : DatabaseTest() {
+    private lateinit var mockClamAv: MockWebServer
 
     @BeforeEach
     fun setup() {
         clearAllMocks()
-        mockWebServer = MockWebServer()
-        mockWebServer.start(6789)
+        mockClamAv = MockWebServer()
+        mockClamAv.start(6789)
     }
 
     @AfterEach
     fun tearDown() {
-        mockWebServer.shutdown()
+        mockClamAv.shutdown()
         checkUnnecessaryStub()
         confirmVerified(cableReportService)
     }
 
-    @Test
-    fun `getMetadataList should return related metadata list`() {
-        val application = alluDataFactory.saveApplicationEntity(USERNAME)
-        attachmentFactory.save(application = application).withDbContent()
-        attachmentFactory.save(application = application).withDbContent()
-
-        val result = applicationAttachmentService.getMetadataList(application.id!!)
-
-        assertThat(result).hasSize(2)
-        assertThat(result).each { d ->
-            d.transform { it.id }.isNotNull()
-            d.transform { it.fileName }.endsWith("file.pdf")
-            d.transform { it.createdByUserId }.isEqualTo(USERNAME)
-            d.transform { it.createdAt }.isSameInstantAs(ApplicationAttachmentFactory.CREATED_AT)
-            d.transform { it.applicationId }.isEqualTo(application.id)
-            d.transform { it.attachmentType }.isEqualTo(MUU)
+    @Nested
+    inner class GetMetadataList {
+        @Test
+        fun `returns empty when application doesn't exist`() {
+            val applicationId = 698432169874L
+
+            val result = applicationAttachmentService.getMetadataList(applicationId)
+
+            assertThat(result).isEmpty()
+        }
+
+        @Test
+        fun `return empty when application doesn't have attachments`() {
+            val application = alluDataFactory.saveApplicationEntity(USERNAME)
+
+            val result = applicationAttachmentService.getMetadataList(application.id!!)
+
+            assertThat(result).isEmpty()
+        }
+
+        @Test
+        fun `returns related metadata list`() {
+            val application = alluDataFactory.saveApplicationEntity(USERNAME)
+            attachmentFactory.save(application = application).withDbContent()
+            attachmentFactory.save(application = application).withDbContent()
+
+            val result = applicationAttachmentService.getMetadataList(application.id!!)
+
+            assertThat(result).hasSize(2)
+            assertThat(result).each { d ->
+                d.prop(ApplicationAttachmentMetadata::id).isNotNull()
```

Comment:
Maybe the d can be dropped and it used instead.

Reply:
Thank you, now using it.